### PR TITLE
Setup CI with AMD platforms

### PR DESCRIPTION
**Pull request by DomAyre**
Todo:

- [ ] Create an agent pool of Azure container instances with SEV-SNP support
- [ ] Setup azure pipeline to run tests using this pool

### CODE REVIEW

File: .azure-pipelines-templates/matrix.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 parameters:
-  target: ["NoSGX", "SGX"]
+  target: ["NoSGX", "SGX", "SNP"]
```
Comment (jumaffre) [inline .azure-pipelines-templates/matrix.yml:2]: Could `NoSGX` be renamed to virtual here and elsewhere as we now have 3 variants?

File: src/host/configuration.h [hunk 2]
```diff
@@ -16,11 +16,13 @@ namespace host
     SGX_RELEASE,
     SGX_DEBUG,
     VIRTUAL,
+    SNP,
```
Comment (jumaffre) [inline src/host/configuration.h:19]: AIUI, this has no effect on the node and is only used by the test infra (`args.enclave_type`)? If that's the case, I think it'd be preferable to have the test infra detect that it's running in a SEV-SNP environment and skip tests accordingly (of course, to be clarified in #4069).

File: .azure-pipelines-templates/matrix.yml [hunk 3]
```diff
@@ -12,6 +12,8 @@ parameters:
     SGX:
       container: sgx
       pool: 1es-dcv2-focal
+    SNP:
+      pool: sev-snp-pool
```
Comment (jumaffre) [inline .azure-pipelines-templates/matrix.yml:16]: Could you add instructions on how to create and configure this pool, even if this eventually subsumed by #4190, in our internal doc repo please?

File: .azure-pipelines-templates/matrix.yml (hunk 1 above)
Comment (DomAyre) [reply .azure-pipelines-templates/matrix.yml:2]: I agree, could be a separate PR?

Comment (achamayou) [inline .azure-pipelines-templates/matrix.yml:2]:
```suggestion
  target: ["NoSGX", "SGX", "SNPCC"]
```

File: .azure-pipelines-templates/matrix.yml (hunk 3 above)
Comment (DomAyre) [reply .azure-pipelines-templates/matrix.yml:16]: https://dev.azure.com/MSRC-CCF/InternalDoc/_wiki/wikis/Internal%20Doc/8/Agent-Pools

File: .azure-pipelines-templates/matrix.yml (hunk 1 above)
Comment (DomAyre) [reply .azure-pipelines-templates/matrix.yml:2]: https://github.com/microsoft/CCF/pull/4197

File: tests/reconfiguration.py [hunk 4]
```diff
@@ -494,10 +497,9 @@ def test_issue_fake_join(network, args):
             assert r.body.json()["node_status"] == ccf.ledger.NodeStatus.PENDING.value
         else:
             assert r.status_code == http.HTTPStatus.UNAUTHORIZED
-            # https://github.com/microsoft/CCF/issues/4072
             assert (
                 r.body.json()["error"]["code"] == "InvalidQuote"
-            ), "SEV-SNP node cannot currently join SGX network"
+            ), "Virtual node must never join SGX network"
```
Comment (jumaffre) [inline tests/reconfiguration.py:502]:
```suggestion
            ), "Virtual node must never join non-virtual network"
```

File: tests/reconfiguration.py [hunk 5]
```diff
@@ -494,10 +497,9 @@ def test_issue_fake_join(network, args):
             assert r.body.json()["node_status"] == ccf.ledger.NodeStatus.PENDING.value
```
Comment (jumaffre) [inline tests/reconfiguration.py:497]: Also check that this is not SEV-SNP

File: tests/code_update.py [hunk 6]
```diff
@@ -95,30 +104,28 @@ def test_update_all_nodes(network, args):
...
 
-    if args.enclave_type == "virtual":
+    if args.enclave_type not in ("release", "debug"):
         # Pretend this was already present
         network.consortium.add_new_code(primary, first_code_id)
 
     LOG.info("Add new code id")
     network.consortium.add_new_code(primary, new_code_id)
     with primary.client() as uc:
+        r = uc.get("/gov/nodes/code_ids")
```
Comment (jumaffre) [inline tests/code_update.py:114]: Remove

File: tests/code_update.py [hunk 7]
```diff
@@ -95,30 +104,28 @@ def test_update_all_nodes(network, args):
...
+                "status": "AllowedToJoin",
+                "platform": "OE_SGX_v1",
+            },
+            {
+                "digest": new_code_id,
+                "status": "AllowedToJoin",
+                "platform": "OE_SGX_v1",
+            },
+        ]
         if args.enclave_type == "virtual":
```
Comment (jumaffre) [inline tests/code_update.py:129]: and not snp

**Review by jumaffre — APPROVED** [review]

**Pull request merged**